packages/labs/src/query-list/queryList.tsx

```diff
@@ -186,7 +186,7 @@ export class QueryList<T> extends React.Component<IQueryListProps<T>, IQueryList
             this.shouldCheckActiveItemInViewport = false;
         }
         // reset active item (in the same step) if it's no longer valid
-        if (this.getActiveIndex() < 0) {
+        if (this.getActiveIndex() < 0 && (this.state.filteredItems.length !== 0 || this.props.activeItem !== undefined)) {
```

Comment:
Much easier to understand. 

It might be prudent to add a bit more to the above comment explaining what this is doing (essentially, don't fire the event if there's nothing to pick and the active item is already undefined) but it's probably fine either way.